File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/HandWagerTest.java
package com.panopset.tests.blackjackEngine;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.blackjackEngine.Bankroll;
import com.panopset.blackjackEngine.Wager;

public class HandWagerTest {

  @Test
  void test() {
    Wager hb = new Wager(new Bankroll(300));
    hb.setInitialBet(500);
    hb.setInitialPayoff(500);
    hb.doubleDown();
    hb.setDoubledPayoff(500);
    Assertions.assertEquals(2000, hb.getLiveValue());
    hb = new Wager(new Bankroll(300));
    hb.setInitialBet(500);
    Assertions.assertEquals(500, hb.getLiveValue());
    hb.lost();
    Assertions.assertEquals(0, hb.getLiveValue());
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/HandWagerTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.Wager
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class HandWagerTest {
    @Test
    fun test() {
        var hb = Wager(500)
        hb.initialBet = 500
        hb.initialPayoff = 500
        hb.doubleDown()
        hb.doubledPayoff = 500
        Assertions.assertEquals(2000, hb.liveValue)
        hb = Wager(300)
        hb.initialBet = 500
        Assertions.assertEquals(500, hb.liveValue)
        hb.lost()
        Assertions.assertEquals(0, hb.liveValue)
    }
}
